upgrade/upgrade-scripts/rootfs-container

```diff
@@ -77,7 +77,17 @@ function get_bootloader_devices() {
 	#
 	zpool list -vH rpool |
 		awk '! /rpool|mirror|replacing|spare/ {print $1}' |
-		sed 's/p\{0,1\}[0-9]*$//'
+		while read -r part; do
+			#
+			# If it doesn't end with a number, it's already a full
+			# disk. Since we expect this to never happen, and the
+			# calling code is likely untested in that case, we
+			# throw an error rather than try to handle it.
+			#
+			[[ "$part" =~ [a-z]$ ]] &&
```

Comment:
I don't feel strongly, but I think this check might be a little more robust if we did something like:
```
[[ "$(lsblk --nodeps -no type "/dev/$part")" == "part" ]] ||
```